src/nvim/api/vim.c

```diff
@@ -3002,3 +3002,23 @@ void nvim_set_decoration_provider(Integer ns_id, DictionaryOf(LuaRef) opts,
 error:
   decor_provider_clear(p);
 }
+
+/// Delete named uppercase/file and numbered mark
+/// @param name       Mark name
+void nvim_del_mark(String name, Error *err)
+  FUNC_API_SINCE(8)
+{
+  if (name.size != 1) {
+    api_set_error(err, kErrorTypeValidation,
+                  "Mark name must be a single character");
+  }
+  // Only allow file/uppercase marks
+  // TODO: Refactor this ASCII_ISUPPER macro to a proper function
+  if (ASCII_ISUPPER(*name.data) || ascii_isdigit(*name.data)) {
+    set_mark(curbuf, name, 0, 0, err);
+  } else {
+    api_set_error(err, kErrorTypeValidation,
+                  "Invalid mark name %c, only file/uppercase marks allowed",
```

Comment:
where possible try to keep the variable part of messages at the end, following a colon. Even if other API functions don't do this yet.
```suggestion
                  "Invalid mark, only file/uppercase marks allowed: %c",
```